### PR TITLE
Fix escaping in ReactDOMInput code

### PR DESCRIPTION
**Pull request by sophiebits**
JSON.stringify isn't the right thing here. Luckily this doesn't look to have any security impact.

### CODE REVIEW

**Review by sebmarkbage — APPROVED** [review]

File: packages/react-dom-bindings/src/client/escapeSelectorAttributeValueInsideDoubleQuotes.js [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+// provided value is going to be wrapped in double quotes as part of an attribute selector
+// Do not use it anywhere else
+// we escape double quotes and backslashes
+const escapeSelectorAttributeValueInsideDoubleQuotesRegex = /[\n\"\\]/g;
+export default function escapeSelectorAttributeValueInsideDoubleQuotes(
+  value: string,
+): string {
+  return value.replace(
+    escapeSelectorAttributeValueInsideDoubleQuotesRegex,
+    ch => '\\' + ch.charCodeAt(0).toString(16),
```
[the remark this answers is not in the document]
Comment (gnoff) [reply packages/react-dom-bindings/src/client/escapeSelectorAttributeValueInsideDoubleQuotes.js:21]: Yeah that seems right

**Review by gnoff — APPROVED** [review]

**Review by gnoff — CHANGES_REQUESTED** [review]
Hmm I thought the space would be consumed by the escape sequence. The selector can’t change, we rely on an invariant that hrefs for float style tags and style sheet link tags do not have spaces in them. This is because we coalesce multiple style tags into a single flushed tag and use this space separated selector to look for a match.


I wouldn’t have thought the escape change would affect this, shouldn’t the new space be consumed?

**Comment by sophiebits** [comment]
(Thanks for looking – was going to tag you on Monday before merging.)

Yes. It's a bug in the selector library used by jsdom that filed an issue for this morning: https://github.com/dperini/nwsapi/issues/84. I confirmed that `~=` with an escape works correctly in Blink, WebKit, and Gecko. I think this new code is correct since I'm still using `data-href~=` but lmk if you don't think so.

**Comment by sophiebits** [comment]
@gnoff thoughts?

**Review by gnoff — APPROVED** [review]
I'm sorry I lost the thread. You're right (of course :))

I would like to refactor against the notion of key <> selector now that they aren't really one to one. But agree this does preserver the style selector semantics

**Comment by sophiebits** [comment]
no worries! thanks for looking

**Pull request merged**